datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -899,23 +898,57 @@ mod test {
 
     #[test]
     fn aggregate() -> Result<()> {
+        #[derive(Debug, Clone)]
```

Comment:
Maybe we could use SimpleAggregateUDF here as well. Not sure if that is any better / worse though